src/pyhf_benchmark/plot.py

```diff
@@ -35,113 +66,75 @@ def load(directory_name):
 
-    return pd.DataFrame.from_dict(output_dic), directory
-
-
-def plot(directory_name):
-    output, directory = load(directory_name)
-    timestamp = output["_runtime"] / 60
-
-    cpu_utilization = output.get("system.cpu", [0] * len(timestamp))
-    disk_utilization = output.get("system.disk", [0] * len(timestamp))
-    proc_cpu_threads = output.get("system.proc.cpu.threads", [0] * len(timestamp))
-    network_sent = output.get("network.sent", [0] * len(timestamp))
-    network_recv = output.get("system.network.recv", [0] * len(timestamp))
-
-    memory = output.get("system.memory", [0] * len(timestamp))
-    proc_memory_availableMB = output.get(
-        "system.proc.memory.availableMB", [0] * len(timestamp)
-    )
-    proc_memory_rssMB = output.get("system.proc.memory.rssMB", [0] * len(timestamp))
-    proc_memory_percent = output.get("system.proc.memory.percent", [0] * len(timestamp))
-
-    gpu_utilization = output.get("system.gpu.0.gpu", [0] * len(timestamp))
-    gpu_memory = output.get("system.gpu.0.memory", [0] * len(timestamp))
-    gpu_memory_alloc = output.get("system.gpu.0.memoryAllocated", [0] * len(timestamp))
-    gpu_temp = output.get("system.gpu.0.temp", [0] * len(timestamp))
-
-    plt.plot(timestamp, cpu_utilization)
-    plt.xlabel("Time(minutes)")
-    plt.ylabel("CPU Utilization (%)")
-    plt.grid()
-    plt.savefig(directory / "CPU_Utilization.png")
-    plt.clf()
-
-    plt.plot(timestamp, disk_utilization)
-    plt.xlabel("Time(minutes)")
-    plt.ylabel("Disk I/O Utilization (%)")
-    plt.grid()
-    plt.savefig(directory / "Disk_IO_Utilization.png")
-    plt.clf()
+    return pd.DataFrame.from_dict(output_dic)
 
-    plt.plot(timestamp, proc_cpu_threads)
-    plt.xlabel("Time(minutes)")
-    plt.ylabel("Process CPU Threads In Use")
-    plt.grid()
-    plt.savefig(directory / "CPU_Threads.png")
-    plt.clf()
-
-    plt.plot(timestamp, network_sent, ls="--")
-    plt.plot(timestamp, network_recv)
-    plt.xlabel("Time(minutes)")
-    plt.ylabel("Network Traffic (bytes)")
-    plt.grid()
-    plt.savefig(directory / "Network_Traffic.png")
-    plt.clf()
 
-    plt.plot(timestamp, memory)
-    plt.xlabel("Time(minutes)")
-    plt.ylabel("System Memory Utilization (%)")
+def load_all(directory_name):
+    list_of_paths = pathlib.Path(directory_name).glob("*")
+    contents = []
+    backends = []
+    for path in list_of_paths:
+        if path.is_dir():
+            backends.append(str(path)[str(path).rfind("_") + 1 :])
+            contents.append(load(path))
+    return contents, backends
+
+
+def subplot(y_label, column, output, directory, filename):
+
+    x_value = output["_runtime"]
+    if y_label == "Network Traffic (bytes)":
+        y_value1 = output.get(column[0], [0] * len(x_value))
+        y_value2 = output.get(column[1], [0] * len(x_value))
+        plt.plot(x_value, y_value1, ls="--", label="send")
+        plt.plot(x_value, y_value2, label="recv")
+        plt.legend(loc="upper left")
```

Comment:
The plotting code should get switched over to using the matplotlib subplots API that I sent you earlier.

Reply:
@matthewfeickert  Can you show me where is your code using matplotlib subplots API? I think I missed it.

Reply:
No. There are lots of examples on using `matplotlib.pyplot.subplots`. Start with the docs https://matplotlib.org/3.3.0/api/_as_gen/matplotlib.pyplot.subplots.html and then there are tons of results on google.

Reply:
@coolalexzb Okay, following up here finally (sorry). What I mean here is that using the `plot.plot` API can be problematic as it is global and so you have to worry about dealing with a universal canvas. It is more robust to use the `plt.subplots` API. So you could simply create a function with a more descriptive name that takes similar arguments but then generates a figure and axis pair from

```python
fig, axis = plt.subplots()
ax.set_xlabel("Time (seconds)")
ax.set_ylabel(y_label)
# ...
return fig, axis
```

and soforth. This is a more robust solution and then actually allows you to _return_ the `fig` and `axis` objects

```python
fig, axis = your_function(y_label, column, output)
# Do other things with fig and axis if you need
fig.savefig(directory / filename)
```